src/main/kotlin/org/ghrobotics/lib/wrappers/TSolenoid.kt

```diff
@@ -0,0 +1,60 @@
+package org.ghrobotics.lib.wrappers
+
+import edu.wpi.first.wpilibj.DoubleSolenoid
+import edu.wpi.first.wpilibj.Solenoid
+
+/**
+ * Interface for both double- and single-solenoids.
+ */
+interface TSolenoid {
+    enum class State {
+        Forward,
+        Reverse,
+        Off
+    }
+
+    var state : State
+}
+
+/**
+ * Single-acting solenoid with only one PCM Solenoid channel.
+ * This type of solenoid can only be on or off.
+ */
+class TSingleSolenoid(channel: Int, module: Int? = null) : TSolenoid {
+    private val wpiSolenoid: Solenoid = if(module == null) Solenoid(channel) else Solenoid(module, channel)
+
+    // Set the solenoid's position. Forward -> true, Reverse -> false
+    // If the solenoid is set to 'Off', the state is not changed.
+    override var state: TSolenoid.State = TSolenoid.State.Reverse
+        set(value) {
+            when (value) {
+                TSolenoid.State.Forward -> wpiSolenoid.set(true)
+                TSolenoid.State.Reverse -> wpiSolenoid.set(false)
+                else -> return
```

Comment:
Wouldn't be an issue with observable, but the field should still be changed if the given state is Off.

Reply:
That's true that the field should be changed. I overlooked that.